File: articles/virtue-ethics-and-the-good-life/fortran/virtue_index.f90
program virtue_index
  implicit none
  real :: virtues(6) = (/4.2, 3.8, 4.4, 4.1, 3.7, 4.0/)
  real :: flourishing(4) = (/4.1, 4.3, 3.9, 4.0/)

  print '(A, F6.3)', 'Mean virtue-domain score: ', sum(virtues) / size(virtues)
  print '(A, F6.3)', 'Mean flourishing score: ', sum(flourishing) / size(flourishing)
end program virtue_index

File: articles/virtue-ethics-and-the-good-life/rust/virtue_index.rs
fn mean(values: &[f64]) -> f64 {
    values.iter().sum::<f64>() / values.len() as f64
}

fn main() {
    let virtues = [4.2, 3.8, 4.4, 4.1, 3.7, 4.0];
    let flourishing = [4.1, 4.3, 3.9, 4.0];

    println!("Mean virtue-domain score: {:.3}", mean(&virtues));
    println!("Mean flourishing score: {:.3}", mean(&flourishing));
}
